lmfit/parameter.py

```diff
@@ -109,7 +109,7 @@ def __deepcopy__(self, memo):
                 param.correl = par.correl
                 param.init_value = par.init_value
                 param.expr = par.expr
-                param.user_data = par.user_data
+                param.user_data = deepcopy(par.user_data)
```

Comment:
I don't have a real issue with doing this, but why is it explicitly needed here and not for any of the other attributes - that doesn't make complete sense to me, but admittedly I haven't looked carefully yet...

Reply:
I am quite new to lmfit, so I'm not sure which of the other attributes are mutable types, maybe `correl` is for example. I agree that it looks a bit asymmetric.